src/cli/main.cpp

```diff
@@ -582,30 +19,36 @@ int main(int argc, char* argv[]) {
        end
      )");
   } catch (const std::exception& e) {
-    rubyInterpreter.evalString(R"(STDOUT.flush)");
+    openstudio::rubyEngine->exec(R"(STDOUT.flush)");
     std::cout << "Exception: " << e.what() << std::endl;  // endl will flush
-    return ruby_cleanup(1);
+    return 1;
   } catch (...) {
-    rubyInterpreter.evalString(R"(STDOUT.flush)");
+    openstudio::rubyEngine->exec(R"(STDOUT.flush)");
     std::cout << "Unknown Exception" << std::endl;  // endl will flush
-    return ruby_cleanup(1);
+    return 1;
   }
-  rubyInterpreter.evalString(R"(STDOUT.flush)");
+
+  openstudio::rubyEngine->exec(R"(STDOUT.flush)");
   std::cout << std::flush;
-  return ruby_cleanup(0);
+  return 0;
 }
 
-extern "C"
-{
-  int rb_hasFile(const char* t_filename) {
-    // TODO Consider expanding this to use the path which we have artificially defined in embedded_help.rb
-    std::string expandedName = std::string(":/ruby/2.7.0/") + std::string(t_filename) + ".rb";
-    return embedded_files::hasFile(expandedName);
-  }
+int main(int argc, char* argv[]) {
+  int result = 0;
 
-  int rb_require_embedded(const char* t_filename) {
-    std::string require_script = R"(require ')" + std::string(t_filename) + R"(')";
-    rubyInterpreter.evalString(require_script);
-    return 0;
+  if ((argc > 1) && (std::string_view(argv[1]) == "experimental")) {
+    CLI::App app{"openstudio"};
+
+    const auto experimentalApp = app.add_subcommand("experimental");
+    [[maybe_unused]] const auto runCommand = experimentalApp->add_subcommand("run");
+
+    CLI11_PARSE(app, argc, argv);
+  } else {
+    result = rubyCLI(argc, argv);
```

Comment:
I have the start of a CLI11 cpp parser at https://github.com/jmarrec/OSCLI11/tree/main/src